python/cudf/cudf/tests/test_scalar.py

```diff
@@ -97,18 +106,20 @@ def test_round_trip_scalar(value):
 @pytest.mark.parametrize(
     "dtype", NUMERIC_TYPES + DATETIME_TYPES + TIMEDELTA_TYPES + ["object"]
 )
-def test_null_scalar(dtype):
-    s = Scalar(None, dtype=dtype)
-    assert s.value is None
+@pytest.mark.parametrize("constructor", [libcudf_scalar, pycudf_scalar])
+def test_null_scalar(dtype, constructor):
```

Comment:
Should we add a similar test for handling `NaT` values as expected?

Reply:
Reworked most of the tests in this file to be appropriate for the new framework